.ruff.toml

```
@@ -0,0 +1,373 @@
+target-version = "py38"
+line-length = 88
+select = ["ALL"]
+exclude=[
+    "astropy/extern/*",
+    "*_parsetab.py",
+    "*_lextab.py"
+]
+extend-ignore = [
+    # ---------------------------------------------------------------
+    # Permanent
+
+    # flake8-builtins (A) : shadowing a Python built-in.
+    # New ones should be avoided and is up to maintainers to enforce.
+    "A00",
+
+    # flake8-bugbear (B)
+    "B008",  # FunctionCallArgumentDefault
+
+    # flake8-commas (COM)
+    "COM812",  # TrailingCommaMissing
+    "COM819",  # TrailingCommaProhibited
+
+    # pydocstyle (D)
+    # Missing Docstrings
+    "D102",  # Missing docstring in public method. Don't check b/c docstring inheritance.
+    "D105",  # Missing docstring in magic method. Don't check b/c class docstring.
+    "D107",  # Missing docstring in __init__. Don't check b/c class docstring.
+    "D213",  # multi-line-summary-second-line
+    # Whitespace Issues
+    "D200",  # FitsOnOneLine
+    "D203",  # OneBlankLineBeforeClass.
+    # Docstring Content Issues
+    "D410",  # BlankLineAfterSection. Using D412 instead.
+    "D400",  # EndsInPeriod. Using D415 instead.
+    "D413",  # BlankLineAfterLastSection.
+    "D416",  # SectionNameEndsInColon.
+
+    # pycodestyle (E, W)
+    "E711",  # NoneComparison  (see unfixable)
+    "E741",  # AmbiguousVariableName | physics variables are often poor code variables
+
+    # Pyflakes (F)
+    "F403",  # ImportStarUsed
+
+    # pep8-naming (N)
+    "N803",  # invalid-argument-name | physics variables are often poor code variables
+    "N806",  # non-lowercase-variable-in-function | physics variables are often poor code variables
+
+    # pandas-vet (PD)
+    "PD",
+
+    # flake8-self (SLF)
+    "SLF001", # private member access
+
+    # Ruff-specific rules (RUF)
+    "RUF005",  # unpack-instead-of-concatenating-to-collection-literal -- it's not clearly faster.
+
+    # ---------------------------------------------------------------
+    # TODO
+
+    # flake8-annotations (ANN) : static typing
+    # TODO: revisit this when Astropy implements typing.
+    "ANN",
+
+    # flake8-unused-arguments (ARG)
+    "ARG001",  # unused-function-argument  # 493 violations (2023-05-03)
```

I don't think writing out these numbers adds much value. Writing out the command to obtain these numbers would be much more useful.
```suggestion
    "ARG001",  # unused-function-argument
```

---

The idea for this is to make it easy for maintainers / contributors to find an issue to work on -- essentially helpful triage. I don't think many / any people will run the command umpteen times before finding something suitable to work on.

---

`ruff --select="ALL" --statistics .` gives much the same information but its output is guaranteed to be up to date with no maintenance burden. Furthermore, some of these rules affect many files and we should encourage addressing them one subpackage at a time. In such cases the numbers listed in the file are not useful at all, but the command can be easily modified, e.g. `ruff --select="ALL" --statistics astropy/time`

---

Fine enough with me. I worry slightly that this is adding a step, reducing the odds someone volunteers to fix something. So far it's mostly been myself and @WilliamJamieson.